src/main/java/uk/gov/pay/adminusers/persistence/dao/UserDao.java

```diff
@@ -93,17 +95,37 @@ public Optional<UserEntity> findByEmail(String email) {
                 .getResultList().stream().findFirst();
     }
 
-    public List<UserEntity> findByServiceId(Integer serviceId) {
+    public List<UserEntity> findByServiceId(Integer serviceId, RoleName roleName) {
 
-        String query = "SELECT s FROM ServiceRoleEntity s " +
-                "WHERE s.service.id = :serviceId ORDER BY s.user.email";
+        String query = "SELECT s FROM ServiceRoleEntity s ";
+        
+        if (roleName != null) {
+            query += " LEFT JOIN RoleEntity re ON s.role = re ";
```

Comment:
It appears you don't have to specify the field in the RoleEntity table because JPA knows how to join it. And it knows how to join it via the information in the [ServiceRoleEntity class](https://github.com/alphagov/pay-adminusers/blob/master/src/main/java/uk/gov/pay/adminusers/persistence/entity/ServiceRoleEntity.java#L31-L32).